File: app/src/main/java/com/jakebarnby/filemanager/models/files/GoogleDriveFile.java
package com.jakebarnby.filemanager.models.files;


import android.net.Uri;

import com.google.api.services.drive.model.File;
import com.jakebarnby.filemanager.util.Constants;

/**
 * Created by Jake on 6/5/2017.
 */

public class GoogleDriveFile extends SourceFile {

    private String mDriveId;

    public GoogleDriveFile(File file) {
        if (file.getWebContentLink() != null) {
            setPath(file.getWebContentLink());
        }
        setDriveId(file.getId());
        setName(file.getName());
        setSourceName(Constants.Sources.GOOGLE_DRIVE);
        setDriveId(file.getId());
        setDirectory(file.getMimeType().equals(Constants.Sources.GOOGLE_DRIVE_FOLDER_MIME));
        setThumbnailLink(file.getHasThumbnail() ? file.getThumbnailLink() : file.getIconLink());
        setModifiedTime(file.getModifiedTime().getValue());
        if (!isDirectory()) {
            setSize(file.getSize() == null? 0 : file.getSize());
        }
        if (file.getWebViewLink() != null) {
            setPath(file.getWebViewLink());
        }
    }
    public String getDriveId() {
        return mDriveId;
    }

    public void setDriveId(String mDriveId) {
        this.mDriveId = mDriveId;
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/models/sources/googledrive/GoogleDriveFile.kt
package com.jakebarnby.filemanager.models.sources.googledrive

import com.google.api.services.drive.model.File
import com.jakebarnby.filemanager.models.SourceFile
import com.jakebarnby.filemanager.models.SourceType
import com.jakebarnby.filemanager.util.Constants.Sources

/**
 * Created by Jake on 6/5/2017.
 */
class GoogleDriveFile(
    file: File
) : SourceFile() {

    init {
        if (file.webContentLink != null) {
            path = file.webContentLink
        }
        if (file.webViewLink != null) {
            path = file.webViewLink
        }

        id = file.id.hashCode().toLong()
        parentFileId = file.parents?.get(0)?.hashCode()?.toLong() ?: -1L
        remoteFileId = file.id.hashCode().toLong()
        name = file.name
        sourceId = SourceType.GOOGLE_DRIVE.id
        isDirectory = file.mimeType == Sources.GOOGLE_DRIVE_FOLDER_MIME

        thumbnailLink = if (file.hasThumbnail) {
            file.thumbnailLink
        } else {
            file.iconLink
        }
        modifiedTime = file.modifiedTime.value
        isHidden = file.name.startsWith(".")

        if (!isDirectory) {
            size = if (file.getSize() == null) {
                0
            } else {
                file.getSize()
            }
        }
    }
}